pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -245,6 +255,44 @@ func (r *ReconcileOneAgent) buildDynatraceClient(instance *dynatracev1alpha1.One
 	return dtc, err
 }
 
+func (r *ReconcileOneAgent) reconcileNodesMarkedForDeletion(reqLogger logr.Logger, dtc dtclient.Client) error {
+
+	// nodes := &corev1.NodeList(metav1.ListOptions{})
+	nodeList := &corev1.NodeList{}
+	fieldSelector := fields.SelectorFromSet(fields.Set{
+		"spec.unschedulable": "True",
+	})
+	listOps := &client.ListOptions{
+		//--field-selector=spec.unschedulable=True
+		FieldSelector: fieldSelector,
+	}
+	err := r.client.List(context.TODO(), listOps, nodeList)
+	if err != nil {
+		reqLogger.Error(err, "failed to list nodes", "listops", listOps)
+		return err
+	}
+
+	for _, node := range nodeList.Items {
+		if node.Spec.Unschedulable {
+			reported, ok := cordonedNodes[&node]
```

Comment:
Since the key is a pointer, different runs of the reconciler may have the same node with different pointers, can't they?

Also, assuming that the pointer remains the same, or we use the name as a key. The scenario: cordon node, uncordon node, cordon node, with the Operator never stopping in the meanwhile, we wouldn't receive the notification for the second cordoning, right?